### PR TITLE
Promote HA VPN to GA

### PR DESCRIPTION
**Pull request by upodroid**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/7534

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:note
compute: added a deprecation warning to `google_compute_vpn_gateway`
```
```release-note:enhancement
compute: promoted `google_compute_ha_vpn_gateway` to GA
```
```release-note:enhancement
compute: promoted `google_compute_external_vpn_gateway` to GA
```
```release-note:enhancement
compute: promoted HA VPN fields in `google_compute_vpn_tunnel` to GA
```

### CODE REVIEW

**Comment by upodroid** [comment]
Tests are passing

```
 REDACTED  MCW0CDP3YY  ~  go  …  github.com  hashicorp  terraform-provider-google   master  60⬇  5✎  1  USAGE  $    make testacc TEST=./google TESTARGS='-run=TestAccComputeHaVpnGateway_haVpnGatewayBasicExample'
==> Checking source code against gofmt...
==> Checking that code complies with gofmt requirements...
go generate  ./...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google -v -run=TestAccComputeHaVpnGateway_haVpnGatewayBasicExample -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google/version.ProviderVersion=acc"
=== RUN   TestAccComputeHaVpnGateway_haVpnGatewayBasicExample
=== PAUSE TestAccComputeHaVpnGateway_haVpnGatewayBasicExample
=== CONT  TestAccComputeHaVpnGateway_haVpnGatewayBasicExample
--- PASS: TestAccComputeHaVpnGateway_haVpnGatewayBasicExample (105.88s)
PASS
ok      github.com/hashicorp/terraform-provider-google/google   109.265s
 REDACTED  MCW0CDP3YY  ~  go  …  github.com  hashicorp  terraform-provider-google   master  62⬇  5✎  4+  1  $    make testacc TEST=./google TESTARGS='-run=TestAccComputeHaVpnGateway_haVpnGatewayGcpToGcpExample'
==> Checking source code against gofmt...
==> Checking that code complies with gofmt requirements...
go generate  ./...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google -v -run=TestAccComputeHaVpnGateway_haVpnGatewayGcpToGcpExample -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google/version.ProviderVersion=acc"
=== RUN   TestAccComputeHaVpnGateway_haVpnGatewayGcpToGcpExample
=== PAUSE TestAccComputeHaVpnGateway_haVpnGatewayGcpToGcpExample
=== CONT  TestAccComputeHaVpnGateway_haVpnGatewayGcpToGcpExample
--- PASS: TestAccComputeHaVpnGateway_haVpnGatewayGcpToGcpExample (285.09s)
PASS
ok      github.com/hashicorp/terraform-provider-google/google   286.919s
```

**Comment by upodroid** [comment]
I also need to mark external_vpn_gateway as GA too.

File: products/compute/api.yaml [hunk 1]
```diff
@@ -14662,7 +14662,6 @@ objects:
     collection_url_key: 'items'
     input: true
     has_self_link: true
-    min_version: beta
     description: |
       Represents a VPN gateway running in GCP. This virtual device is managed
       by Google, but used only by you. This type of VPN Gateway allows for the creation
```
Comment (jtymes) [inline products/compute/api.yaml:14667]: Should the `api` reference on line 14674 be updated to from `https://cloud.google.com/compute/docs/reference/rest/beta/vpnGateways` to `https://cloud.google.com/compute/docs/reference/rest/v1/vpnGateways`?

Comment (upodroid) [reply products/compute/api.yaml:14667]: Oh, that the link to the docs. I'll fix that.

**Comment by ndmckinley** [comment]
You'll need to disable this one for Ansible - you want to add an `exclude: true` to this resource in ansible.yaml.

**Review by ndmckinley — APPROVED** [review]
I'm seeing the unit tests pass, so LGTM.

**Pull request merged**